### PR TITLE
exec: report cpu usage summary stat

### PR DESCRIPTION
**Pull request by notnoop**
Fix cases where some fields don't get serialized across grpc calls: namely CPU TotalTicks and Memory Swaps.  I added a test for round trip checking.

Looking at the `nomad alloc status --stats`, this makes two changes:
* CPU reporting is no longer always zero
* `Memory Stats` includes Swap reporting.

note that this includes Memory Swap in reporting as well

```
==> Monitoring evaluation "0bb58439"
    Evaluation triggered by job "test_raw"
    Allocation "371e6389" created: node "72e3d5c2", group "test"
    Allocation "371e6389" status changed: "pending" -> "running" (Tasks are running)
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "0bb58439" finished with status "complete"
mars-2:nomad notnoop$ nomad alloc status --stats 371e6389
ID                  = 371e6389
Eval ID             = 0bb58439
Name                = test_raw.test[0]
Node ID             = 72e3d5c2
Job ID              = test_raw
Job Version         = 0
Client Status       = running
Client Description  = Tasks are running
Desired Status      = run
Desired Description = <none>
Created             = 9s ago
Modified            = 8s ago

Task "test1" is "running"
Task Resources
CPU           Memory          Disk     Addresses
2246/100 MHz  25 MiB/300 MiB  300 MiB

Memory Stats
RSS     Swap
25 MiB  0 B

CPU Stats
Percent  System Mode  User Mode
102.10%  1.98%        100.11%

Task Events:
Started At     = 2019-03-30T19:23:02Z
Finished At    = N/A
Total Restarts = 0
Last Restart   = N/A

Recent Events:
Time                       Type        Description
2019-03-30T15:23:02-04:00  Started     Task started by client
2019-03-30T15:23:02-04:00  Task Setup  Building Task Directory
2019-03-30T15:23:02-04:00  Received    Task received by client```
```

### CODE REVIEW

File: plugins/drivers/utils.go [hunk 1]
```diff
@@ -441,6 +441,11 @@ func resourceUsageToProto(ru *ResourceUsage) *proto.TaskResourceUsage {
 			cpu.MeasuredFields = append(cpu.MeasuredFields, proto.CPUUsage_THROTTLED_TIME)
 		case "Percent":
 			cpu.Percent = ru.CpuStats.Percent
+
+			// total ticks is derived from percent, and should always be set together
+			// even if "Total Ticks" is not marked as an explicit measured field
```
Comment (notnoop) [inline plugins/drivers/utils.go]: FWIW, I tried adding Total Ticks as an explicit measured field, and got the following alloc status output:

```
CPU Stats
Percent  System Mode  Total Ticks  User Mode
101.12%  0.00%        101.12%
```

So I opted to make it implicit and preserve 0.8 behavior as much as possible.

**Comment by preetapan** [comment]
I still see `TotalTicks` set to zero in the `v1/client/allocation/<id>/stats` endpoint and in the CLI output of `nomad alloc status -verbose <id>` in this branch. e.g

```
 "ResourceUsage": {
        "CpuStats": {
          "Measured": [
            "System Mode",
            "User Mode",
            "Percent"
          ],
          "Percent": 358.9639876722877,
          "SystemMode": 0,
          "ThrottledPeriods": 0,
          "ThrottledTime": 0,
          "TotalTicks": 0,
          "UserMode": 358.96399020429607
        },
```
My test job is a `raw_exec` job running stress . 0.8.7 returns non zero values for `TotalTicks` for the same job.

**Comment by notnoop** [comment]
Oops - I forgot to commit the deserializer change!  Can you test again?

**Comment by notnoop** [comment]
After reviewing rest of code, noticed that we are dropping Swap usage metric as well, so updated PR and added a round-trip test.

**Comment by notnoop** [comment]
Thanks for the e2e test!

**Pull request merged**